src/stdune/code_error.ml

```diff
@@ -1,14 +1,15 @@
 type t =
   { message : string
   ; data : (string * Dyn.t) list
+  ; loc : Loc0.t option
   }
 
 exception E of t
 
-let raise message data =
-  raise (E { message; data })
+let raise ?loc message data =
+  raise (E { message; data ; loc })
 
-let to_dyn { message; data } : Dyn.t =
+let to_dyn { loc = _; message; data } : Dyn.t =
```

Comment:
I think I would expect `to_dyn` to keep the location information, for cases where we collect multiple code_errors into a larger data structure (I'm vaguely thinking the Memo backtrace thing, but I don't really remember what that does), and to have `to_dyn_drop_loc` for use in the case where we report the loc explicitly.

Reply:
I'll include it. I actually tried pretty printing the loc properly here, but got tangled up in circular deps. Printing the loc as `Dyn.t` is fine for now.